static/app/components/events/eventStatisticalDetector/eventDifferentialFlamegraph.tsx

```diff
@@ -33,13 +35,13 @@ export function EventDifferenialFlamegraph(props: EventDifferenialFlamegraphProp
   }, [isValid, fingerprint, breakpoint]);
 
   const projectID = parseInt(props.event.projectID, 10);
+  const transactions = useTransactionsDelta();
 
-  // @todo use the data to render a flamegraph
   useDifferentialFlamegraphQuery({
     projectID,
     breakpoint,
     environments: [],
-    transaction: '',
+    transaction: transactions.data?.data?.[0]?.transaction as string,
```

Comment:
The transaction isn't always defined so i think you need a way to wait for it to load by disabling this query until the transaction name is ready

Reply:
Yeah, you cant tell this just from this change, but we already do that. useDifferentialFlamegraphQuery just delegates to two useAggregateFlamegraphQuery calls which disables it [here](https://github.com/getsentry/sentry/blob/master/static/app/utils/profiling/hooks/useAggregateFlamegraphQuery.ts#L31)